pandas/core/tools/datetimes.py

```diff
@@ -254,6 +254,16 @@ def _convert_listlike(arg, box, format, name=None, tz=tz):
                 require_iso8601 = not infer_datetime_format
                 format = None
 
+        if format is not None:
```

Comment:
can you move this check to

``pandas/_libs/tslibs/strptime.pyx``

its just closer to the code where we use it